examples/draw_shuffling_graph.py

```diff
@@ -0,0 +1,88 @@
+import time
+import mxnet as mx
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+from gluonts.dataset.loader import TrainDataLoader
+from gluonts.model.deepar import DeepAREstimator
+from gluonts.dataset.artificial import shuffle_testing_dataset
+
+BATCH_SIZE = 32
+NUM_BATCH_PER_EPOCH = 50
+range_list = range(1, 30)
+
+
+def plot_heap_map(c, num_worker, subplot, shuffle):
+    n = 300
+    if shuffle:
+        shuffle_buffer_length = c * BATCH_SIZE
+    else:
+        shuffle_buffer_length = None
+    train_ds = shuffle_testing_dataset(n)
+    estimator = DeepAREstimator(
+        prediction_length=20,
+        freq="D",
+    )
+    transform = estimator.create_transformation()
+    loader = TrainDataLoader(
+        train_ds,
+        transform=transform,
+        batch_size=BATCH_SIZE,
+        ctx=mx.cpu(),
+        num_batches_per_epoch=NUM_BATCH_PER_EPOCH,
+        shuffle_buffer_length=shuffle_buffer_length,
+        num_workers=num_worker,
+    )
+    hit_list = [[0 for i in range(NUM_BATCH_PER_EPOCH)] for j in range(n)]
+    print(f"dataset size: {len(train_ds)}")
+    start_time = time.time()
+    count = 0
+    for batch in loader:
+        print(f"item id: {batch['item_id']}")
+        for item_id in batch['item_id'][0]:
+            hit_list[item_id][count] += 1
+        count += 1
+    end_time = time.time()
+    print(f"elapsed time: {end_time - start_time}")
+    if subplot:
+        sns.set()
+        ax = sns.heatmap(hit_list)
+        title = "buffer length={} * batch_size, {} process".format(c,
+        num_worker) if shuffle_buffer_length is not None else "No.{} not shuffle, {} process".format(
+            c, num_worker)
+        plt.title(title)
+        plt.show()
+    return end_time - start_time
+
+
+def generate_time_lst(num_worker=None, subplot=False, shuffle=True):
+    time_list = []
+    for c in range_list:
+        t = plot_heap_map(c, num_worker, subplot, shuffle)
+        time_list.append(t)
+    return time_list
+
+
+def compare():
+    n_worker = 2
+    subplot = False
+    single_process_time_shuffle = generate_time_lst(subplot=subplot)
+    multi_worker_time_shuffle = generate_time_lst(num_worker=n_worker, subplot=subplot)
+    single_process_time_not_shuffle = generate_time_lst(shuffle=False, subplot=subplot)
+    multi_worker_time_not_shuffle = generate_time_lst(num_worker=n_worker, shuffle=False, subplot=subplot)
+    plt.plot(range_list, single_process_time_shuffle, label="single process shuffle")
+    plt.plot(range_list, multi_worker_time_shuffle, label="{} process shuffle".format(n_worker))
+    plt.plot(range_list, single_process_time_not_shuffle, label="single process not shuffle")
+    plt.plot(range_list, multi_worker_time_not_shuffle, label="{} process not shuffle".format(n_worker))
+    plt.legend()
+    plt.show()
+
+
+def length_vs_randomness(n, n_worker=None):
+    for i in range(1, n):
+        plot_heap_map(i, n_worker, subplot=True, shuffle=True)
+
+
+if __name__ == '__main__':
```

Comment:
This script is really nice, but doesn't need to be merged; could you instead edit the original post of the PR and add there plots showing results (heatmaps, running times) obtained with this?

Reply:
Where should I add these plots?

Reply:
on the top post in this issue, you can edit that and add code snippets or images

Reply:
I've uploaded the graphs.